testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1836,6 +1837,33 @@ public void testDoubleUpdateAndThenReadFromHive()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL)
+    public void testDeleteWithOriginalFiles()
+    {
+        withTemporaryTable("test_delete_with_no_predicate_on_ctas", true, false, NONE, tableName -> {
+            onTrino().executeQuery("SET SESSION scale_writers = true"); //these 3 properties are necessary to make sure there is more than 1 original file created
+            onTrino().executeQuery("SET SESSION writer_min_size = '4kB'");
+            onTrino().executeQuery("SET SESSION task_writer_count = 2");
+            onTrino().executeQuery(format(
+                    "CREATE TABLE %s WITH (transactional = true) AS SELECT * FROM tpch.sf1000.orders LIMIT 100000", tableName));
+
+            verify(onTrino().executeQuery(format("SELECT DISTINCT \"$path\" FROM %s", tableName)).getRowsCount() >= 2,
+                    "There should be at least 2 files");
+            validateFileIsDirectlyUnderTableLocation(tableName);
+
+            onTrino().executeQuery(format("DELETE FROM %s", tableName));
```

Comment:
Add another test, with DELETE with condition, e.g. `WHERE orderkey % 2 = 0`. 
It's probably easier to delete all rows "correctly" than to delete "many rows".

writing verification queries can be challenging, because `tpch.sf1000.orders LIMIT 100000` may be non-deterministic.  will be simpler, if you use the `tiny` schema (see comment above)

Reply:
I think it is still possible to go with limit if I add ordering

Reply:
ordering seems to be way too slow..

Reply:
it should be fine with the way verification is done now

Reply:
> ordering seems to be way too slow..

especially if you use a large schema 
that's why i didn't suggest ordering in the first place